cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.api;
+
+import co.cask.cdap.api.DatasetConfigurer;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.data.stream.StreamSpecification;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+import co.cask.cdap.internal.dataset.DatasetCreationSpec;
+import com.google.common.base.Preconditions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of {@link DatasetConfigurer} for adding datasets and streams in Configurers.
+ */
+public class DefaultDatasetConfigurer implements DatasetConfigurer {
```

Comment:
@chtyim True, but the ``DefaultFlowletConfigurer`` is in cdap-api which depends on ``DefaultDatasetConfigurer``. ``DefaultFlowConfigurer`` is in cdap-api since ``FlowletDefinition`` needs it. And ``FlowletDefinition`` is in cdap-api since the old configurer style ``FlowSpecification`` needs it. So for now I have placed it in cdap.internal.api. Will move it to cdap-app-fabric after removing old Flow configurer API.